common/tools/dev-tool/src/commands/run/typecheck.ts

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import * as path from "node:path";
+import { leafCommand, makeCommandInfo } from "../../framework/command";
+import { createPrinter } from "../../util/printer";
+import { resolveProject } from "../../util/resolveProject";
+import { Project } from "ts-morph";
+import { DiagnosticCategory } from "typescript";
+
+const log = createPrinter("typecheck");
+
+export const commandInfo = makeCommandInfo(
+  "typecheck",
+  "typecheck typescript code files that are not part of tshy build",
+  {
+    paths: {
+      kind: "string",
+      description:
+        "relative path pattern of additional files (not included in tsconfig.json) to check for compilation errors. Examples: '--paths samples-dev/**/*.ts'.",
```

Comment:
maybe not relevant this, but since tshy would ignore the test and samples-dev folder i was wondering if we should not include them into tsconfig.json by default.
https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/mongocluster/arm-mongocluster/tsconfig.json#L8

Reply:
Either way is fine with me since this command allows passing patterns.